DBZ-8412: Support for Postgres 17 failover slots;

gunnarmorling opened the pull request:
Creating Postgres replication slots enabled for failover when both of the following conditions are true:

- Postgres version is >= 17
- Debezium is connecting to a primary (only primaries support failover slots right now)

Updating the docs to reflect the support for failover slots, as well as to clarify that replication slots on stand-by servers are supported with Postgres 16 or newer.

Fixes https://issues.redhat.com/browse/DBZ-8412

gunnarmorling commented on debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java:
```diff
@@ -368,6 +368,22 @@ private boolean useTemporarySlot() throws SQLException {
+    private boolean useFailOverSlot() throws SQLException {
+        boolean supportsFailOverSlots = pgConnection().haveMinimumServerVersion(ServerVersion.from("17"));
+
+        AtomicBoolean isPrimary = new AtomicBoolean();
```
Good idea, will do that 👍 .

jpechane commented:
Can we aim for https://issues.redhat.com/browse/DBZ-6410?focusedId=26124743&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-26124743 ?

rk3rn3r commented:
> @gunnarmorling great addition. Even it will be complicated to have an integration tests to tests the failover, maybe an example like the [MySQL](https://github.com/debezium/debezium-examples/tree/main/failover) will be great to have.

Maybe it is not that complicated. https://github.com/debezium/debezium/blob/main/debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresShutdownIT.java might help implementing a failover integration test suite, wdyt? @gunnarmorling @mfvitale

gunnarmorling commented:
Reworked a bit, this should be good to go now. Added an option for enabling this explicitly. Also added a test to verify the slot is configured as it should. This should be good enough, we don't need to test that Postgres itself handles slot failover correctly.

gunnarmorling commented:
Woot, all tests passing now 🥳 .

mfvitale commented:
> Maybe it is not that complicated. https://github.com/debezium/debezium/blob/main/debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresShutdownIT.java might help implementing a failover integration test suite, wdyt? @gunnarmorling @mfvitale

Not impossible but not immediate. We have https://issues.redhat.com/browse/DBZ-6410 to track the missing test scenario.

jpechane commented:
@gunnarmorling Applied, thanks a lot! We'd still appreciate additional testing provided with [DBZ-6410](https://issues.redhat.com/browse/DBZ-6410) if you have a spare cycles for it.
